core/modules/filter/filter.module

```diff
@@ -1342,12 +1342,21 @@ function filter_process_format($element) {
       '#parents' => array_merge($element['#parents'], array('format')),
     );
   }
+  // If there's only a single text format available to the current user, then
+  // render it as a hidden form element.
   else {
     $element['format']['format'] = array(
       '#type' => 'hidden',
       '#value' => $element['#format'],
       '#default_value' => $element['#format'],
-      '#attributes' => array('class' => array('filter-list')),
+      '#attributes' => array(
+        'class' => array('filter-list'),
+        // This hidden input element contains only the machine name of the text
+        // format (as its value). We need the human-readable name of the text
+        // format, to be shown as a tooltip/summary in the "Formating options"
+        // fieldset; so we pass that as a data-* attribute.
```

Comment:
I think this could be simplified:

```suggestion
        // Add the human-readable name of the text format so it can be used as
        // the fieldset summary.
```